### PR TITLE
Feature/fix arround oidc spec

### PR DESCRIPTION
**Pull request by mokemokechicken**
Fix for pass test of http://openid.net/certification/testing/

* HTTP Response of TokenEndpoint should return application/json Content-Type
* fix at_hash computation in IdToken.

### CODE REVIEW

File: src/OAuth2/OpenID/ResponseType/IdToken.php [hunk 1]
```diff
@@ -6,6 +6,7 @@
 use OAuth2\Encryption\Jwt;
 use OAuth2\Storage\PublicKeyInterface;
 use OAuth2\OpenID\Storage\UserClaimsInterface;
+use Phalcon\Logger;
```
Comment (copynpaste) [inline src/OAuth2/OpenID/ResponseType/IdToken.php]: Remove this pls

Comment (mokemokechicken) [inline src/OAuth2/OpenID/ResponseType/IdToken.php]: I'm sorry. I deleted it!

**Comment by bshaffer** [comment]
This is great! Thank you for finding this and fixing these.

File: src/OAuth2/Controller/TokenController.php [hunk 2]
```diff
@@ -51,7 +51,8 @@ public function handleTokenRequest(RequestInterface $request, ResponseInterface
             // server MUST disable caching in headers when tokens are involved
             $response->setStatusCode(200);
             $response->addParameters($token);
-            $response->addHttpHeaders(array('Cache-Control' => 'no-store', 'Pragma' => 'no-cache'));
+            $response->addHttpHeaders(array('Cache-Control' => 'no-store', 'Pragma' => 'no-cache',
+                'Content-Type' => 'application/json'));
```
Comment (casperbiering) [inline src/OAuth2/Controller/TokenController.php]: This change should not be needed, since it's set when calling send method on Reponse object.

Comment (bshaffer) [inline src/OAuth2/Controller/TokenController.php]: How would it know what content type to use unless we set it?

Comment (casperbiering) [inline src/OAuth2/Controller/TokenController.php]: We _do_ set it. When calling $response->send();
https://github.com/bshaffer/oauth2-server-php/blob/develop/src/OAuth2/Response.php#L203

Comment (bshaffer) [inline src/OAuth2/Controller/TokenController.php]: That's only implemented in my response class, not as part of the interface.

**Comment by bshaffer** [comment]
closing in favor of #698 (rebased and whitespace cleanup)

**Pull request closed without merging**